src/Test/Utilities/Portable/PDB/DeterministicBuildCompilationTestHelpers.cs

```diff
@@ -0,0 +1,179 @@
+using Xunit;
+
+namespace Roslyn.Test.Utilities.PDB
+{
+    internal static partial class DeterministicBuildCompilationTestHelpers
+    {
+        public static IEnumerable<EmitOptions> GetEmitOptions()
+        {
+            var emitOptions = new EmitOptions(
+                debugInformationFormat: DebugInformationFormat.Embedded,
+                pdbChecksumAlgorithm: HashAlgorithmName.SHA256,
+                defaultSourceFileEncoding: Encoding.UTF32);
+
+            yield return emitOptions;
+            yield return emitOptions.WithDefaultSourceFileEncoding(Encoding.ASCII);
+            yield return emitOptions.WithDefaultSourceFileEncoding(null).WithFallbackSourceFileEncoding(Encoding.UTF7);
+            yield return emitOptions.WithFallbackSourceFileEncoding(Encoding.UTF7).WithDefaultSourceFileEncoding(Encoding.ASCII);
+        }
+
+        internal static void AssertCommonOptions(EmitOptions emitOptions, CompilationOptions compilationOptions, Compilation compilation, ImmutableDictionary<string, string> pdbOptions)
+        {
+            if (emitOptions.FallbackSourceFileEncoding != null)
+            {
+                Assert.Equal(emitOptions.FallbackSourceFileEncoding.WebName, pdbOptions["fallback-encoding"]);
+            }
+
+            if (emitOptions.DefaultSourceFileEncoding != null)
+            {
+                Assert.Equal(emitOptions.DefaultSourceFileEncoding.WebName, pdbOptions["default-encoding"]);
+            }
+
+            var runtimeInformationalVersionAttribute = (AssemblyInformationalVersionAttribute)typeof(object).Assembly.GetCustomAttributes(typeof(AssemblyInformationalVersionAttribute), false).Single();
+            Assert.Equal(runtimeInformationalVersionAttribute.InformationalVersion, pdbOptions["runtime-version"]);
+
+            int portabilityPolicy = 0;
+            if (compilationOptions.AssemblyIdentityComparer is DesktopAssemblyIdentityComparer identityComparer)
+            {
+                portabilityPolicy |= identityComparer.PortabilityPolicy.SuppressSilverlightLibraryAssembliesPortability ? 0b1 : 0;
+                portabilityPolicy |= identityComparer.PortabilityPolicy.SuppressSilverlightPlatformAssembliesPortability ? 0b10 : 0;
+            }
+
+            Assert.Equal(portabilityPolicy.ToString(), pdbOptions["portability-policy"]);
+
+            var compilerVersion = typeof(Compilation).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion;
+            Assert.Equal(compilerVersion.ToString(), pdbOptions["compiler-version"]);
+
+            var runtimeVersion = typeof(object).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion;
+            Assert.Equal(runtimeVersion, pdbOptions["runtime-version"]);
+
+            var optimization = compilationOptions.OptimizationLevel == OptimizationLevel.Debug
+                ? (compilationOptions.DebugPlusMode ? "debug+" : "debug")
+                : "release";
```

Comment:
This is the third place this logic is used.

Reply:
Resolving, discussed in https://github.com/dotnet/roslyn/pull/44373#discussion_r434905122

Reply:
This is only used in a single place in production code now, right?

---
In reply to: [435471517](https://github.com/dotnet/roslyn/pull/44373#discussion_r435471517) [](ancestors = 435471517)